proto-lens-protoc/src/Data/ProtoLens/Setup.hs

```diff
@@ -77,7 +95,29 @@ defaultMainGeneratingProtos root
 generatingProtos
     :: FilePath -- ^ The root directory under which .proto files can be found.
     -> UserHooks -> UserHooks
-generatingProtos root hooks = hooks
+generatingProtos root = generatingSpecificProtos root getProtos
+  where
+    getProtos p = do
+      -- Replicate Cabal's own logic for parsing file globs.
+      files <- concat <$> mapM matchFileGlob (extraSrcFiles p)
+      pure $ filter (isSubdirectoryOf root)
+           $ filter (\f -> takeExtension f == ".proto")
+               files
+
+-- | Augment the given 'UserHooks' to auto-generate Haskell files from the
+-- .proto files returned by a function @getProtos@.
+--
+-- Writes the generated files to the autogen directory (@dist\/build\/autogen@
+-- for Cabal, and @.stack-work\/dist\/...\/build\/autogen@ for stack).
+--
+-- Throws an exception if the @proto-lens-protoc@ executable is not on the PATH.
+generatingSpecificProtos
```

Comment:
I feel there's a mismatch between `generatingSpecificProtos` which takes a `PackageDescription -> IO [FilePath]` callback and `defaultMainGeneratingSpecificProtos` which just takes a `[FilePath]`.  Compare for example:

    defaultMainGeneratingSpecificProtos :: (PackageDescription -> IO [FilePath]) -> IO ()

What do you think about that API?  Or renaming one of the functions to make the asymmetry more clear?